components/server/src/workspace/workspace-starter.ts

```diff
@@ -543,6 +546,35 @@ export class WorkspaceStarter {
         }
     }
 
+    private async checkStartPermission(user: User, workspace: Workspace, project?: Project) {
+        // explicit project
+        if (project) {
+            return;
+        }
+
+        const { organizationId, contextURL } = workspace;
+
+        const membership = await this.orgDB.findTeamMembership(user.id, organizationId);
+        if (!membership) {
+            return;
+        }
+
+        // check if user's role is restricted from starting arbitrary repositories
+        const organizationSettings = await this.orgService.getSettings(user.id, organizationId);
+        if (!organizationSettings?.roleRestrictions?.[membership.role]?.includes("start_arbitrary_repositories")) {
+            return;
+        }
+
+        // implicit project (existing on the same clone URL)
+        const projects = await this.projectService.findProjectsByCloneUrl(user.id, contextURL, organizationId);
+        if (projects.length === 0) {
+            throw new ApplicationError(
+                ErrorCodes.PRECONDITION_FAILED,
+                "Unable to start workspace: This repository has not been imported. Your role is restricted to using only imported repositories for workspace creation. Please contact your organization owner to import this repository or modify permissions.",
```

Comment:
> Please contact your organization owner to import this repository or modify permissions.

Org members are allowed to create project as well

Reply:
@mustard-mh do you have a suggestion for a better message? Thanks for pointing out the discrepancy, it would be nice to capture that, but I can't think of anything that would sound good (like "import this repository if you can" sounds weird, same as "contact an organization owner or member").

Reply:
Let's left it as it's now